app/cli.py

```diff
@@ -162,3 +171,37 @@ def create_tables():
     @algolia.command()
     def reindex():
         reindex_all()
+
+    @apikeys.command()
+    @click.argument('apikey')
+    def blacklist(apikey):
```

Comment:
I [mentioned before](https://github.com/OperationCode/resources_api/issues/225#issuecomment-546477867) the assumption that each user would only ever have one API key. In that world, this method alone would suffice.

In a world in which users have multiple API keys, we're also going to want the ability to blacklist a user altogether. So all of their API keys are invalid, and if they try to request a new one from the `/apikey` or `/new_apikey` endpoint, they would be rejected. Blacklisting a user should only be a click command, not available through the public API.

It doesn't look like you've implemented multiple API keys per user, and if you don't, this is perfectly fine for now. However, I could still see it useful to pass in an email address of a user and blacklist their one API key, so I think that if you're fine with it, I'd like you to add this `blacklist(email)` method as well. You can probably reuse much of the same logic that you have in this method, you just need to pull the key from the database differently.

Reply:
Was thinking the parameter should allow you to use email or key. I will work on that.

For the future, I'm not sure if we need a world with n number of keys, but just 2 keys since the only real use case is key rotation. In that case, we could just add another field like `previous_key` (if populated, then that key also works, if it's `null`, it's ignored). WDYT?

Reply:
Right, probably just two keys, but from what I understand, blacklisting doesn't delete a key, right? So if you rotate your key yearly, then after 2 years you'd have more than 2 API keys. So maybe you only have two active at any given time, but there would still be more than 2 associated with an account so we'd want them all to stay blacklisted.